### PR TITLE
Feature products service

### PR DESCRIPTION
**Pull request by jejikeh**

### CODE REVIEW

File: src/Versity.ApiGateway/Versity.ApiGateway/Program.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption;
+using Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel;
+using Ocelot.DependencyInjection;
+using Ocelot.Middleware;
+using Versity.ApiGateway.Extensions;
+
+var builder = WebApplication.CreateBuilder(args);
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.ApiGateway/Versity.ApiGateway/Program.cs]: Pleeeease, format it))))))))

File: src/Versity.Products/Core/Application/Abstractions/Repositories/IVersityProductsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Domain.Models;
+
+namespace Application.Abstractions.Repositories;
+
+public interface IVersityProductsRepository
+{
+    public IQueryable<Product> GetAllProducts();
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/Abstractions/Repositories/IVersityProductsRepository.cs:7]: Hm, queryable to bll?

File: src/Versity.Products/Core/Application/ApplicationInjection.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+public static class ApplicationInjection
+{
+    public static IServiceCollection AddApplication(this IServiceCollection serviceCollection)
+    {
+        serviceCollection.AddScoped(typeof(IPipelineBehavior<,>), typeof(LoggingPipelineBehavior<,>));
+        serviceCollection.AddScoped(typeof(IPipelineBehavior<,>), typeof(ValidationPipelineBehavior<,>));
+        serviceCollection.AddValidatorsFromAssembly(Assembly.GetExecutingAssembly());
+        serviceCollection.AddMediatR(configuration => configuration.RegisterServicesFromAssemblies(Assembly.GetExecutingAssembly()));
+        return serviceCollection;
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/ApplicationInjection.cs:18]: :)

File: src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsRequestHandler.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        _products = products;
+    }
+
+    public Task<Result<IEnumerable<Product>>> Handle(GetAllProductsQuery request, CancellationToken cancellationToken)
+    {
+        var products = _products
+            .GetAllProducts()
+            .OrderBy(x => x.Title)
+            .Skip(10 * (request.Page - 1))
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsRequestHandler.cs]: I think i saw it

File: src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        app.UseHttpsRedirection();
+        app.UseAuthentication();
+        app.UseAuthorization();
+        app.UseCors("AllowAll");
+        app.MapControllers();
+        
+        return app;
+    }
+    
+    private static IServiceCollection AddJwtAuthentication(this IServiceCollection serviceCollection, IConfiguration configuration)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs]: Extension method

File: src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        });
+
+        builder.Services.AddCors(options => options.AddPolicy("AllowAll", policy =>
+        {
+            policy.AllowAnyHeader();
+            policy.AllowAnyMethod();
+            policy.AllowAnyOrigin();
+        }));
+        
+        builder.Host.UseSerilog((context, configuration) =>
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs]: Extension method

File: src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                ValidIssuer = Environment.GetEnvironmentVariable("JWT__Issuer") ?? configuration.GetSection("Jwt:Issuer").Value,
+                ValidAudience = Environment.GetEnvironmentVariable("JWT__Audience") ?? configuration.GetSection("Jwt:Audience").Value,
+                IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(Environment.GetEnvironmentVariable("JWT__Key") ?? configuration.GetSection("Jwt:Key").Value))
+            };
+        });
+
+        return serviceCollection;
+    }
+    
+    private static ElasticsearchSinkOptions ConfigureElasticSink(IConfigurationRoot configuration, string enviroment)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs]: Extension method

File: src/Versity.Products/External/Presentation/Program.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+var builder = WebApplication
+    .CreateBuilder(args)
+    .ConfigureBuilder();
+
+var app = builder
+    .Build()
+    .ConfigureApplication();
+
+using var scope = app.Services.CreateScope();
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/External/Presentation/Program.cs:13]: format it

File: src/Versity.Products/Core/Application/Abstractions/Repositories/IVersityProductsRepository.cs (hunk 2 above)
Comment (jejikeh) [reply src/Versity.Products/Core/Application/Abstractions/Repositories/IVersityProductsRepository.cs:7]: I thought it was OK, so as not to pull the entire database for myself, instead I request a certain number of entities [here](https://github.com/jejikeh/versity/pull/17/files#diff-c541299cef0b1131fb90a236db4b3e261721e87e807b0204674bba2f9ebbd481R23)

File: src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsRequestHandler.cs (hunk 4 above)
Comment (jejikeh) [reply src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsRequestHandler.cs]: the same way i do [in this command handler](https://github.com/jejikeh/versity/pull/17/files#diff-4b98c05264f69a3e6d1e1c8850465d3d8c5993b2f4e85b65d42c497b6099cbb5L7)

File: src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs (hunk 5 above)
Comment (jejikeh) [reply src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs]: I didn't quite understand what you meant here, but I put the method data in a [separate class and file](https://github.com/jejikeh/versity/pull/17/files#diff-53bf72a090c03bb959c8336bb56b8f0c42d668dc9af9f66b9a9edddf0508c40f)

File: src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs (hunk 6 above)
Comment (jejikeh) [reply src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs]: I didn't quite understand what you meant here, but I put the method data in a [separate class and file](https://github.com/jejikeh/versity/pull/17/files#diff-6e8fd5248f660e8a2b4f8716f6d87f1b8d71a90379d24438ed5df6f0f9f6c06a)

File: src/Versity.ApiGateway/Versity.ApiGateway/Program.cs (hunk 1 above)
Comment (jejikeh) [reply src/Versity.ApiGateway/Versity.ApiGateway/Program.cs]: [here :|](https://github.com/jejikeh/versity/pull/17/files#diff-1d2d8482ab2784404f04d7d82928adf85555f5284fbefaff59f3021c248cb8eeR4)

File: src/Versity.Products/External/Presentation/Program.cs (hunk 8 above)
Comment (jejikeh) [reply src/Versity.Products/External/Presentation/Program.cs:13]: I don't understand how I should format it, except like this
```c#
using var scope = app.Services
    .CreateScope();
```
this okay?

**Pull request merged**